src/Nodes1DProvisioner.cpp

```diff
@@ -159,61 +160,43 @@ namespace blitzdg {
     }
 
     void Nodes1DProvisioner::buildConnectivityMatrices() {
-        firstIndex ii;
-        secondIndex jj;
-        thirdIndex kk;
-
-        index_type totalFaces = NumFaces*NumElements;
+        index_type totalFaces = NumFaces * NumElements;
         index_type numVertices = NumElements + 1;
+        index_type localVertNum[2] = {0, 1};
 
-        index_type localVertNum[2];
-        localVertNum[0] = 0; localVertNum[1] = 1;
-
-        // Build global face-to-vertex array. (should be sparse matrix in 2D/3D).
-        real_matrix_type FToV(totalFaces, numVertices);
-        FToV = 0*jj;
-
-        index_matrix_type & E2V = *EToV;
-
+        // Build global face-to-vertex array. Note: we actually
+        // build its transpose for easy column access.
+        CSCMat FToVtrans(numVertices, totalFaces, totalFaces);
+        const index_matrix_type& E2V = *EToV;
         index_type globalFaceNum = 0;
-        for (index_type k=0; k < NumElements; k++) {
-            for (index_type f=0; f < NumFaces; f++) {
+        for (index_type k = 0; k < NumElements; ++k) {
+            for (index_type f = 0; f < NumFaces; ++f) {
                 index_type v = localVertNum[f];
-                index_type vGlobal = E2V(k,v);
-                FToV(globalFaceNum, vGlobal) = 1;
-                globalFaceNum++;
+                index_type vGlobal = E2V(k, v);
+                FToVtrans.colPtrs(globalFaceNum) = globalFaceNum;
+                FToVtrans.rowInds(globalFaceNum) = vGlobal;
+                FToVtrans.elems(globalFaceNum++) = 1.0;
```

Comment:
I appreciate the conciseness of this, but I worry about the readability by combining the counter increment with the array assignment here.

Can we separate it into two lines again?

Reply:
Sure thing Frank.